### PR TITLE
fixed boundary conditions top/bottom for uvfg

### LINKED ISSUE #53: Boundaries of Boundary Conditions
adapt the `applyBoundaryConditions` for top and bottom to only set the boundaries of the cells in the computational domain. currently the are also set in the halo and ghost cells. 

Similar problem as in #52

### LINKED ISSUE #55: Calculate Number of Subdomains in Partitioning
Currently, we assume that we divide the computational domain in 2x2 subdomains. 

Write a function that computes `nSubdomains`and `mSubdomains` in Partitioning

### PR DESCRIPTION
**Pull request by kimkroener**
fixes #53

### CODE REVIEW

File: src/computation/1_computation_parallel.cpp [hunk 1]
```diff
@@ -274,56 +274,33 @@ void ComputationParallel::applyBoundaryValuesRight()
...
-    {
-        discretization_->u(i, discretization_->uJBegin())   = 2. * settings_.dirichletBcBottom[0] - discretization_->u(i, discretization_->uJBegin() +1);
-    }
     
-    // v
-    for ( int i = 0; i < discretization_->vIEnd(); i++)
+    for ( int i = 0; i < discretization_->nCells[0]; i++)
     {
+        // u 
+        discretization_->u(i, discretization_->uJBegin())  = 2. * settings_.dirichletBcBottom[0] - discretization_->u(i, discretization_->uJBegin() +1);
```
Comment (MagnusOstertag) [inline src/computation/1_computation_parallel.cpp]: ich will die Indizes, zum besser verstehen

File: src/partitioning/partitioning.cpp [hunk 2]
```diff
@@ -54,3 +49,34 @@ nCellsGlobal_(nCells)
...
+        {
+            m = nRanks_ / n;
+            cost = nCellsGlobal_[0]*(m-1) + nCellsGlobal_[1]*(n-1); // number of innner edges with partition n x m
+            
+            if ( cost < cost_opt) {
+                n_opt = n; 
+                m_opt = m; 
+                cost_opt = cost;
+            }
+        };
```
Comment (JanisWissinger) [inline src/partitioning/partitioning.cpp]: I think the semicolons aren't  needed after the if and for brace

**Review by JanisWissinger — APPROVED** [review]

File: src/partitioning/partitioning.cpp [hunk 3]
```diff
@@ -6,14 +6,9 @@ nRanks_(nRanks),
...
     // map ownRank -> subdomain
-
-    // TODO: more than two rows
-    // factorize N
-    // minimize cost for each factorization
-
-    // only for two rows
-    int n_subd = 2; // number of subdomains in i direction or nCells[0]
-    int m_subd = 2; // number of subdomains in j direction or nCells[1]
+    int n_subd = nSubdomains_[0]; // number of subdomains in i direction or nCells[0]
```
Comment (MagnusOstertag) [inline src/partitioning/partitioning.cpp:11]: you first have to call factorize()

**Review by MagnusOstertag — APPROVED** [review]

File: src/partitioning/partitioning.cpp (hunk 2 above)
Comment (kimkroener) [reply src/partitioning/partitioning.cpp]: fixed

File: src/partitioning/partitioning.cpp (hunk 3 above)
Comment (kimkroener) [reply src/partitioning/partitioning.cpp:11]: fixed

**Pull request merged**